datascience/src/pipeline/queries/fmc/historic_controls.sql

```diff
@@ -0,0 +1,101 @@
+WITH T1 AS (
+
+SELECT
+rc.id_fmc_bc_resultat_ctrl_env as id,
+rc.ID_FMC_BC_MISSION as mission_id,
+rc.NBRE_PERS_NAV_CTRLES as action_number_of_controls,
+
+REGEXP_SUBSTR(inf.LIBELLE_AUTRE_INFRACTION,  '\d{4,5}') as lib_au_inf,
+nat.CODE as code,
+nat.LIBELLE as libelle_nat,
+rc.DATE_DEBUT as action_start_datetime_utc,
+cnc.LIBELLE as actionTheme,
+cep.LIBELLE as protected_species
+
+
+FROM FMC2.FMC_BC_RESULTAT_CTRL_ENV rc
+
+LEFT JOIN FMC2.FMC_BC_RES_CTRL_ENV_INFR inf
+ON inf.ID_FMC_BC_RESULTAT_CTRL_ENV=rc.ID_FMC_BC_RESULTAT_CTRL_ENV
+
+LEFT JOIN FMC2.FMC_CODE_NATINF nat
+ON inf.IDC_FMC_NATINF=nat.IDC_FMC_NATINF
+
+INNER JOIN FMC2.FMC_BC_RES_CTRL_ENV_NAT rcn
+ON rc.ID_FMC_BC_RESULTAT_CTRL_ENV=rcn.ID_FMC_BC_RESULTAT_CTRL_ENV
+
+INNER JOIN FMC2.FMC_CODE_NATURE_CONTROLE cnc
+ON cnc.IDC_FMC_NATURE_CONTROLE=rcn.IDC_FMC_NATURE_CONTROLE
+
+LEFT JOIN FMC2.FMC_BC_ESPECE_PROTEGEE ep
+ON ep.ID_FMC_BC_PERTURBATION=rc.ID_FMC_BC_PERTURBATION
+
+LEFT JOIN FMC2.FMC_CODE_ESPECE_PROTEGEE cep
+ON ep.IDC_FMC_ESPECE_PROTEGEE=cep.IDC_FMC_ESPECE_PROTEGEE),
+
+T2 as (
+SELECT
+id,
+action_number_of_controls,
+protected_species,
+LISTAGG(lib_au_inf,',') WITHIN GROUP (ORDER BY id) as extracted_nat,
+mission_id,
+actionTheme,
+LISTAGG(code,',') WITHIN GROUP (ORDER BY id) as natinf,
+LISTAGG(libelle_nat, ',') WITHIN GROUP (ORDER BY id) as libelle_nat,
+action_start_datetime_utc
+
+FROM T1
+GROUP BY
+id,
+protected_species,
+mission_id,
+action_number_of_controls,
+action_start_datetime_utc,
+actionTheme),
+
+
+T3 as (
+SELECT
+T2.id,
+CONCAT(T2.id,T2.action_start_datetime_utc) as temp_id,
+CASE
```

Comment:
Deux remarques :
- dans le `CONCAT`, il faut plutôt utiliser `','` comme séparateur, et non `', '`, comme c'est fait partout ailleurs, sinon les espaces se retrouvent dans les natinf
- si `T2.extracted_nat IS NULL` et `T2.NATINF is NULL` simultanément, alors le résultat est `NULL` -> dans les test data, dans ton `historic_controls_df` il faut qu'il y ait des `None` pour couvrir le cas des contrôles sans infraction, et il n'y en a pas.